src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -72,6 +88,13 @@ private void RunWorker()
 
                 try
                 {
+#if TARGET_OSX
+                    if (!string.IsNullOrEmpty(_thread.Name))
+                    {
+                        // Name the underlying native thread to match the managed thread name.
+                        _thread.ThreadNameChanged(_thread.Name);
```

Comment:
Can you just call CurrentThread here instead of passing the thread around manually? It should reduce ifdefs significantly.